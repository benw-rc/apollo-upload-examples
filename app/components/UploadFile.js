import gql from 'graphql-tag'
import { graphql } from 'react-apollo'

const S3_DOCUMENT_STORE_BUCKET = 'my-bucket'
const S3_DOCUMENT_STORE_KEY = 'reqs'

let s3Location = {}
s3Location = {
  bucket: S3_DOCUMENT_STORE_BUCKET,
  key: S3_DOCUMENT_STORE_KEY
}

const UPLOAD_FILE = gql`
  mutation singleUpload($input: DocumentInput!) {
    singleUpload(input: $input) {
      id
      fileName
      uploadedBy
      uploadedAt
    }
  }
`

const uploadOneFile = ({ mutate }) => {
  const handleChange = ({
    target: {
      validity,
      files: [file]
    }
  }) => {
    //eslint-disable-next-line
    console.log('iupload: ', file)
    const input = {
      config: {
        AMPLICON: 'amplicon',
        EMAILS: 'al@rc.com'
      },
      file,
      s3Location
    }

    return (
      validity.valid &&
      mutate({
        variables: { input }
      })
    )
  }

  return <input type="file" required onChange={handleChange} />
}

const enhancedUploadOneFile = graphql(UPLOAD_FILE)
const uploadFile = enhancedUploadOneFile(uploadOneFile)

export default uploadFile
