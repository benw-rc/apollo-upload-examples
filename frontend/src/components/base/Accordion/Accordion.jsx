/** @module */
import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import { Accordion as BaseAccordion } from 'semantic-ui-react';

import './Accordion.scss';

/**
 * @callback select_cb
 * @param {Object} e - event object
 * @param {Object} - clicked title element data
 */

/**
 * Simple Accordion overlay that adds our theming
 * Accepts all options for the semantic-ui-react Accordion
 *
 * <https://react.semantic-ui.com/modules/accordion>
 *
 * @param {string=} className - additional css classes
 * @param {select_cb=} on_select - callback to called w/ -1 if the element
 *   is active, or the index of the selected title card otherwise.
*/
export const Accordion = ({ className, on_select, ...props }) =>
  <BaseAccordion
    className={classNames('argus-accordion', className)}
    styled
    onTitleClick={(e, { index, active }) => on_select( active ? -1 : index )}
    inverted
    {...props}
  />

Accordion.propTypes = {
  className: PropTypes.string,
  on_select: PropTypes.func,
}
Accordion.defaultProps = {
  on_select: () => ({}),
};

export default Accordion;
