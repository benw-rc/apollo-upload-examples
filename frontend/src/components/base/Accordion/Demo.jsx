import React from 'react';

import Accordion from './Accordion';

const LabelComponent = ({ value }) => <div>{value}</div>;

export const Demo = ({ className, ...props }) => {
  const mk_panel = (title, content) => ({ key: title, title, content: { content } });
  return (
    <div>
      <Accordion
        className='test-example'
        panels={[
          { key: 1, title: 'Basic', content: { content: "An Entry" } },
          {
            key: 2,
            title: 'Component',
            content: { content: <LabelComponent value="Label Component" /> }
          },
          mk_panel('Short-hand', <LabelComponent value="Short-Hand Component" />),
        ]}
      />
    </div>
  );
}

export default Demo;
