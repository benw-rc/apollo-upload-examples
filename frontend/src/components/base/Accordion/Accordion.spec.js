import _ from 'lodash';

import { render, simple_test } from 'rc-modules/redux/test/components';
import { Accordion } from './Accordion';

describe('Accordion component', () => {
  const props = { some: 'props', for_the: 'tests' };
  const className = 'cln';
  simple_test(
    'basic',
    Accordion,
    {
      className,
      on_select: jest.fn(),
      ...props,
    }
  );
  describe('prop tests', () => {
    let el, sub_el, el_props, on_select;

    beforeEach(() => {
      on_select = jest.fn();
      el = render(Accordion, {
        className,
        on_select,
        ...props,
      });
      sub_el = el.find('Accordion');
      el_props = sub_el.props();
    });

    it('forwards props', () => {
      expect(el_props.className.split(' ')).toContain(className);
      _.mapValues(props, (val) => expect(el_props[val]).toEqual(props[val]));
    });

    it('calls on_select on title click (index if inactive, -1 if active)', () => {
      const index = 2;
      el_props.onTitleClick('err', { index, active: true });
      expect(on_select).toHaveBeenCalledWith(-1);
      el_props.onTitleClick('err', { index, active: false });
      expect(on_select).toHaveBeenCalledWith(index);
    });
  });

});
