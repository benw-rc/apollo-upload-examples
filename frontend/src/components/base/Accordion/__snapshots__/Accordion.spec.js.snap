// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion component :basic 1`] = `
<Accordion
  className="argus-accordion cln"
  for_the="tests"
  inverted={true}
  onTitleClick={[Function]}
  some="props"
  styled={true}
/>
`;
